model/k8s/metrics.yaml

```diff
@@ -469,3 +469,200 @@ groups:
+    instrument: gauge
+    unit: "By"
+    note: |
+      More info: https://kubernetes.io/docs/concepts/policy/resource-quotas/#compute-resource-quota
+
+      This metric SHOULD, at a minimum, be reported against a
+      [`k8s.resourcequota`](../resource/k8s.md#resourcequota) resource.
+    attributes:
+      - ref: k8s.resourcequota.state
+        requirement_level: required
+  - id: metric.k8s.resourcequota.memory.request
+    type: metric
+    metric_name: k8s.resourcequota.memory.request
+    stability: development
+    brief: |
+      The memory requests in a specific namespace.
+      Across all pods in a non-terminal state,
+      the sum of memory requests cannot exceed this value.
+    instrument: gauge
+    unit: "By"
+    note: |
+      More info: https://kubernetes.io/docs/concepts/policy/resource-quotas/#compute-resource-quota
+
+      This metric SHOULD, at a minimum, be reported against a
+      [`k8s.resourcequota`](../resource/k8s.md#resourcequota) resource.
+    attributes:
+      - ref: k8s.resourcequota.state
+        requirement_level: required
+
+  # k8s.resourcequota.hugepages.* metrics
+  - id: metric.k8s.resourcequota.hugepages.request
+    type: metric
+    metric_name: k8s.resourcequota.hugepages.request
+    stability: development
+    brief: |
+      The huge page requests in a specific namespace.
+      Across all pods in a non-terminal state,
+      the sum of huge page requests cannot exceed this value.
+    instrument: gauge
+    unit: "By"
+    note: |
+      More info: https://kubernetes.io/docs/concepts/policy/resource-quotas/#compute-resource-quota
+
+      This metric SHOULD, at a minimum, be reported against a
+      [`k8s.resourcequota`](../resource/k8s.md#resourcequota) resource.
+    attributes:
+      - ref: k8s.resourcequota.state
+        requirement_level: required
+      - ref: k8s.hugepage.size
+        requirement_level: required
+
+  # k8s.resourcequota.storage.* metrics
+  - id: metric.k8s.resourcequota.storage.request
+    type: metric
+    metric_name: k8s.resourcequota.storage.request
+    stability: development
+    brief: |
+      The storage requests in a specific namespace.
+
+      Across all persistent volume claims associated with the <storage-class-name> (if exists),
+      the sum of storage requests cannot exceed this value.
+    instrument: gauge
+    unit: "By"
+    note: |
+      More info: https://kubernetes.io/docs/concepts/policy/resource-quotas/#storage-resource-quota
+
+      This metric SHOULD, at a minimum, be reported against a
+      [`k8s.resourcequota`](../resource/k8s.md#resourcequota) resource.
+    attributes:
+      - ref: k8s.resourcequota.state
+        requirement_level: required
+      - ref: k8s.storage.class_name
+        requirement_level: opt_in
+
+  # k8s.resourcequota.persistentvolumeclaims.* metrics
+  - id: metric.k8s.resourcequota.persistentvolumeclaims.request
+    type: metric
+    metric_name: k8s.resourcequota.persistentvolumeclaims.request
```

Comment:
There is no request/limits for persistentvolumeclaims (the storage.* quotas are for quantity). The field `persistentvolumeclaims` in resourcequota is just the count of pvcs permitted or used. We can either make this part of `object_count` with `k8s.storage.class_name` a conditional_required attr when `k8s.resourcequota.resource_name = pvc` OR we just drop the `.request` and change the unit on this to `1`

Reply:
Good catch! Thank you!